packages/core/src/connectivity/scp-cf/destination/destination-service.spec.ts

```diff
@@ -486,16 +470,32 @@ describe('destination service', () => {
         .get('/destination-configuration/v1/destinations/FINAL-DESTINATION')
         .reply(400, response);
 
-      fetchDestination(destinationServiceUri, jwt, destinationName, {
-        enableCircuitBreaker: false
-      })
-        .then(() => {
-          done("Should've failed...");
+      await expect(() =>
+        fetchDestination(destinationServiceUri, jwt, destinationName, {
+          enableCircuitBreaker: false
         })
-        .catch(error => {
-          expect(error.message).toBeDefined();
-          done();
-        });
+      ).rejects.toThrowError();
+    });
+
+    it('circuit breaker opens after 10 failed request attempts', async () => {
+      const attempts = 10;
```

Comment:
This magic number comes from the hard coded option `volumeThreshold`, maybe reuse that value.
Also, I know the other 3 options were there for long time, could we also add some tests as you are enabling the circuit breaker.

Reply:
Good point, I will adjust to volumeThreshold.

I tried to make a test for the reset and thresholdpercentage, but due to an [issue](https://github.com/facebook/jest/issues/10221) with jest fake timers and nock, I was not able to do this. (Took me a long time already unfortunately).
Anyways, I think testing that the circuit breaker is in fact used is important, while checking that the circuit breaker behaves as we expect should not necessarily be our business. I think, here it would make sense, but the benefit does not justify the cost.

Reply:
As discussed, partially this was tested manually, but due to the jest / nock combination issue it is currently not easily possible to automatically test this.